### PR TITLE
[fix](nullable) Fix nullable cache to avoid function returning wrong value

### LINKED ISSUE #14407: [Bug] aggregate result not correct
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

latest-master

### What's Wrong?

1. create test table:
```
CREATE TABLE `t3` (
  `k1` int(11) NULL,
  `a1` int(11) NULL
) ENGINE=OLAP
DUPLICATE KEY(`k1`)
COMMENT 'OLAP'
DISTRIBUTED BY HASH(`k1`) BUCKETS 10
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"in_memory" = "false",
"storage_format" = "V2",
"disable_auto_compaction" = "false"
);
```

2. insert test data
`insert into t3 values(1,1),(2,2),(3,3),(4,null),(null,5);`

3. wrong result with SQL:
```
> select min(k1) from (select k1,a1 from t3 union select null,null) t;
+-----------+
| min(`k1`) |
+-----------+
|         0 |
+-----------+
```

4. correct min(k1) should be 1, such as:
```
> select min(k1) from (select k1 from t3 union select null) t;
+-----------+
| min(`k1`) |
+-----------+
|         1 |
+-----------+
```

### What You Expected?

+-----------+
| min(`k1`) |
+-----------+
|         1 |
+-----------+

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yukang-Lian**
# Proposed changes

Issue Number: close #14407 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vaggregation_node.h [hunk 1]
```diff
@@ -278,6 +278,7 @@ struct AggregationMethodKeysFixed {
                 ColumnNullable& nullable_col = assert_cast<ColumnNullable&>(*key_columns[i]);
                 observed_column = &nullable_col.get_nested_column();
                 null_map = assert_cast<ColumnUInt8*>(&nullable_col.get_null_map_column());
```
[the remark this answers is not in the document]
Comment (Yukang-Lian) [reply be/src/vec/exec/vaggregation_node.h]: done.

**Review by cambyzju — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.29 seconds
 load time: 437 seconds
 storage size: 17170774967 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221123082845_clickbench_pr_51432.html

**Comment by An-DJ** [comment]
It seems that the failed regression test `regression (ARM pipeline)` was caused by the internal error of TeamCity [agent](http://43.132.222.7:8111/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=51436&_focus=1868). (virtual memory exhausted: Cannot allocate memory).

Maybe it can be solved by triggering the CI‘s rebuild process.

**Comment by Yukang-Lian** [comment]
> It seems that the failed regression test `regression (ARM pipeline)` was caused by the internal error of TeamCity [agent](http://43.132.222.7:8111/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=51436&_focus=1868). (virtual memory exhausted: Cannot allocate memory).
> 
> Maybe it can be solved by triggering the CI‘s rebuild process.

Thanks for your advice! TeamCity agents are not always stable. However, I don't have permission to rebuild it so I think I need to ask someone to do it for me.

**Comment by yiguolei** [comment]
This issue is related with https://github.com/apache/doris/pull/13289.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**